### DISCUSSION TITLE
Only include column with new formatting if displayed in table?

### DISCUSSION THREAD

**Discussion by uqjohart** [pxlrbt/filament-excel#166] [Q&A] [answered]
OK so I am sure there is a simple way to do this... any help greatly appreciated. 
I have a column "staff.username" that I am messing with the formatting in the table and I want to setup the bulk export to format it differently but only include it if it is displayed in thetable ( toggleable() ).

Any idea how to fix that? Here is the export code:

               ExportBulkAction::make()
                    ->exports([
                        ExcelExport::make()
                            ->fromTable()
                                ->withColumns([
                                    Column::make('staff.username')
                                        ->formatStateUsing(fn (string $state): string => __($state." EXTRA FORMATTING NOT IN TABLE")),
                        ])
                    ]),

**Comment by pxlrbt** [comment]
Just checked the `CanToggleColumns` source code and I think this could work:

```php
Column::make('staff.username')
   ->formatStateUsing(fn (string $state): string => __($state." EXTRA FORMATTING NOT IN TABLE"))
   ->hidden(fn ($livewire) => $livewire->isTableColumnToggledHidden('staff.username'))
```

**Reply by uqjohart** [reply]
You are a king among kings, will give it a crack first thing in the morning. Greatly appreciate it. I'll mark it as answered once in place.

**Reply by uqjohart** [reply]
Hmm "Hmm... Call to undefined method pxlrbt\FilamentExcel\Columns\Column::hidden()"
Doesnt look like that hidden function exists even in the repo.

**Reply by uqjohart** [reply, marked as answer]
Woho fixed it! I have to mess with the withClumns but that let me use the $livewire call you provided thanks :D


```
->withColumns(function ($livewire) {
    $returnedArray = [];
    $displayed = $livewire->isTableColumnToggledHidden('staff.username');
    if (!$displayed) $returnedArray[] = Column::make('staff.username');
    return $returnedArray;
})
```
